plugins/lsp/language-servers/default.nix

```
@@ -146,27 +148,52 @@ let
           default = true;
           example = false;
         };
-      };
-      extraConfig = cfg: opts: {
-        assertions = lib.nixvim.mkAssertions "plugins.lsp.servers.vue_ls" {
-          assertion = cfg.tslsIntegration -> (cfg.package != null);
-          message = "When `${opts.tslsIntegration}` is enabled, `${opts.package}` must not be null.";
+        vtslsIntegration = mkOption {
+          type = types.bool;
+          description = ''
+            Enable integration with vtsls.
+          '';
+          default = true;
+          example = false;
         };
-        plugins.lsp.servers.ts_ls = lib.mkIf (cfg.enable && cfg.tslsIntegration) {
-          filetypes = [ "vue" ];
-          extraOptions = {
-            init_options = {
-              plugins = lib.mkIf (cfg.package != null) [
-                {
-                  name = "@vue/typescript-plugin";
-                  location = "${lib.getBin cfg.package}/lib/language-tools/packages/language-server";
-                  languages = [ "vue" ];
-                }
+      };
+      extraConfig =
+        cfg: opts:
+        let
+          plugin = {
+            name = "@vue/typescript-plugin";
+            location = "${lib.getBin cfg.package}/lib/language-tools/packages/language-server";
+            languages = [ "vue" ];
+          };
+        in
+        {
+          assertions = lib.nixvim.mkAssertions "plugins.lsp.servers.vue_ls" {
+            assertion = (cfg.tslsIntegration or cfg.vtslsIntegration) -> (cfg.package != null);
+            message = "When `${opts.tslsIntegration}` or `${opts.vtslsIntegration}` is enabled, `${opts.package}` must not be null.";
+          };
+          plugins.lsp.servers.ts_ls = lib.mkIf (cfg.enable && cfg.tslsIntegration) {
```

```suggestion
          plugins.lsp.servers.ts_ls = lib.mkIf cfg.tslsIntegration {
```

Everything in `extraConfig` is already conditioned on `cfg.enable`:
https://github.com/nix-community/nixvim/blob/a1d3ca973234f8013fb99903e915b700e6ddbbb5/lib/plugins/mk-neovim-plugin.nix#L93-L107